<?xml version="1.0"?>
<launch>
    <arg name="dev" default="/dev/roboclaw"/>
    <arg name="baud" default="115200"/>
    <arg name="frontaddr" default="128"/>
    <arg name="backaddr" default="129"/>
    <arg name="max_speed" default="2.0"/>
    <arg name="ticks_per_meter" default="104000"/>
    <arg name="base_width" default="0.92"/>
    <arg name="linear_max_vel" default="2.0"/>
    <arg name="angular_max_vel" default="2.0"/>
    <arg name="accel" default="100000"/> <!-- roboclaw default is 655360 -->

    <!-- <rosparam file="$(find roboclaw_ros)/config/control.yaml" command="load"/> -->
    <arg name="respawn" default="true" />
    <arg name="respawn_delay" default="2" />
    <arg name="run_diag" default="false" />

    <node pkg="roboclaw_serial_ros" type="roboclaw_node.py" name="roboclaw_node" respawn="$(arg respawn)" respawn_delay="$(arg respawn_delay)" output="screen">
        <param name="~dev" value="$(arg dev)"/>
        <param name="~baud" value="$(arg baud)"/>
        <param name="~frontaddr" value="$(arg frontaddr)"/>
        <param name="~backaddr" value="$(arg backaddr)"/>
        <param name="~max_speed" value="$(arg max_speed)"/>
        <param name="~ticks_per_meter" value="$(arg ticks_per_meter)"/>
        <param name="~base_width" value="$(arg base_width)"/>
        <param name="~linear/x/max_velocity" value="$(arg linear_max_vel)"/>
        <param name="~angular/z/max_velocity" value="$(arg angular_max_vel)"/>
        <param name="~accel" value="$(arg accel)"/>
    </node>
<!--
    <node if="$(arg run_diag)" pkg="diagnostic_aggregator" type="aggregator_node"
          name="diagnostic_aggregator">
        <rosparam command="load"
                  file="$(find roboclaw_ros)/config/roboclaw_diag.yaml"/>
    </node>
-->
</launch>
